app.go

```diff
@@ -299,6 +308,108 @@ func NewApp(
 	return app, nil
 }
 
+// InitAppForTestnet executes the necessary state transitions on the
+// provided application in order to start an in-place testnet.
+func InitAppForTestnet(app *App, pubKey crypto.PubKey, rawConsensusAddress bytes.HexBytes, rawOperatorAddress string, upgradeToTrigger string) *App {
+	ctx := app.NewUncachedContext(true, cmtproto.Header{})
+
+	// ===== Cosmos SDK: Staking  =====
+
+	cmtPubKey := &ed25519.PubKey{Key: pubKey.Bytes()}
+	operatorAddress := sdk.MustBech32ifyAddressBytes("noblevaloper", sdk.MustAccAddressFromBech32(rawOperatorAddress))
+	validator, err := stakingtypes.NewValidator(
+		operatorAddress,
+		cmtPubKey,
+		stakingtypes.Description{Moniker: "Testnet Validator"},
+	)
+	if err != nil {
+		cmtos.Exit("failed to create new validator: " + err.Error())
```

Comment:
Is there anything special about this vs. using a regular `os.Exit` call?

Reply:
It allows us to both print the message and `os.Exit(1)`, which cleans up the code slightly.

Let me know if you want me to update this in a follow on PR!